generator/src/generator.rs

```diff
@@ -17,12 +17,28 @@ use pest::unicode::unicode_property_names;
 use pest_meta::ast::*;
 use pest_meta::optimizer::*;
 
-pub fn generate(
+#[derive(Debug)]
+pub(crate) struct DocComment<'a> {
+    pub(crate) grammar_docs: Vec<&'a str>,
+    pub(crate) line_docs: Vec<Vec<&'a str>>,
+    pub(crate) rules: Vec<Rule>,
+}
+
+impl DocComment<'_> {
+    fn line_docs_for_rule(&self, rule_name: &str) -> Option<String> {
+        let idx = self.rules.iter().position(|r| r.name == rule_name)?;
+
+        self.line_docs.get(idx).map(|comments| comments.join("\n"))
```

Comment:
this may increase the compilation time for larger grammars a bit. So, my suggestion would be:
1. instead of constructing DocComment directly, make those inner fields private and have a `new` constructor function that takes pairs etc. and have that construction functionality (`consume_grammar_doc`, `consume_line_docs`) there instead of doing the construction externally (so that `DocComment` can maintain invariants about those inner fields)
2. not sure if there's any other purpose for `rules` other than looking up the index by the rule name? If no, maybe it'll be easier (and faster) just to have e.g. `line_docs: HashMap<String, Vec<&'a str>>`, so that line docs can be looked up directly?

Reply:
Done